src/components/scenes/WalletRestoreScene.tsx

```diff
@@ -0,0 +1,185 @@
+
+interface Props extends EdgeSceneProps<'walletRestore'> {}
+
+export function WalletRestoreScene(props: Props) {
+  const { navigation } = props
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const account = useSelector(state => state.core.account)
+  const allKeys = useWatch(account, 'allKeys')
+  const restoreWalletInfos = allKeys.filter(key => key.archived || key.deleted).reverse()
+
+  const [searchText, setSearchText] = React.useState('')
+  const [selectedWalletInfos, setSelectedWalletInfos] = React.useState<EdgeWalletInfoFull[]>([])
+
+  const handlePressDone = useHandler(async () => {
+    await dispatch(showRestoreWalletsModal(navigation, selectedWalletInfos))
+  })
+
+  const handleSearchClear = useHandler(() => {
+    setSearchText('')
+  })
+
+  const handleSelectionChanged = useHandler((walletInfo: EdgeWalletInfoFull, isSelected: boolean) => {
+    if (isSelected) {
+      setSelectedWalletInfos([...selectedWalletInfos, walletInfo])
+    } else {
+      setSelectedWalletInfos(selectedWalletInfos.filter(selectedWalletInfo => selectedWalletInfo.id !== walletInfo.id))
+    }
+  })
+
+  return (
+    <SceneWrapper avoidKeyboard>
+      {({ insetStyle }) => (
+        <View style={styles.container}>
+          <SceneHeaderUi4 title={lstrings.restore_wallets_modal_title} />
+          <SimpleTextInput
+            horizontalRem={0.5}
+            bottomRem={0.5}
+            returnKeyType="search"
+            placeholder={lstrings.search_wallets}
+            onChangeText={setSearchText}
+            onClear={handleSearchClear}
+            value={searchText}
+            iconComponent={SearchIconAnimated}
+          />
+          <RestoreList insetStyle={insetStyle} restoreWalletInfos={restoreWalletInfos} searchText={searchText} onSelectionChanged={handleSelectionChanged} />
+          <SceneButtons
+            primary={{
+              label: lstrings.restore,
+              onPress: handlePressDone,
+              disabled: selectedWalletInfos.length === 0
+            }}
+            absolute
+          />
+        </View>
+      )}
+    </SceneWrapper>
+  )
+}
+
+const RestoreList = (props: {
+  insetStyle: InsetStyle
+  restoreWalletInfos: EdgeWalletInfoFull[]
+  searchText: string
+  onSelectionChanged: (walletInfo: EdgeWalletInfoFull, isSelected: boolean) => void
+}) => {
+  const { insetStyle, restoreWalletInfos, searchText, onSelectionChanged } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const currencyConfigMap = useSelector(state => state.core.account.currencyConfig)
+
+  const filteredWalletInfos = React.useMemo(() => {
+    const searchTarget = normalizeForSearch(searchText)
+
+    const out: Array<EdgeWalletInfoFull | string> = []
+    let lastSection = ''
+    for (const restoreWalletInfo of restoreWalletInfos) {
+      const { type, created } = restoreWalletInfo
+      const pluginId = type.replace('wallet:', '')
```

Comment:
I'm noticing a pattern here. Maybe we need a helper function to get the currencyConfig for a `walletInfo.type`:

```ts
function findCurrencyConfig(account: EdgeAccount, walletType: string): EdgeCurrencyConfig | undefined {
  for (const pluginId of Object.keys(account.currencyConfig)) {
    const config = account.currencyConfig[pluginId]
    if (config.currencyInfo.walletType === walletType) return config
   }
}
```

This would be used not only here, but also earlier in the `WalletRestoreCard`.